### DISCUSSION TITLE
Identify resolved value source for a particular key

### DISCUSSION THREAD

**Discussion by kristijorgji** [firebase/flutterfire#6707] [Feature request]
Currently we can get values like this
```
_remoteConfig.getBool('test');
```

and also set defaults before fetch and activate
```
_remoteConfig.setDefaults({
'test': false
});
```

In my result of _remoteConfig.getBool('test') I will receive false when no value is configured in remote config ui for key `test`, that is correct, however for many purposes we need to know if the false is a default value or is set false in remote config.

Currently I see no way how to check something like `_remoteConfig.keyExists('test')`

Or like in firebase remote config library for other languages, where is nicely possible like this
```
val rawValue = remoteConfig.getValue(key)
val exists = rawValue.source == FirebaseRemoteConfig.VALUE_SOURCE_REMOTE
```

How can i do the same for the dart flutter library ?, so check if source is the remote config or the default value i set, or the type default value string '' bool false and so on

We have build wrapper against firebase remote config to allow more granular specification of defaults, not everything in one place as have huge configs and this will make it possible.

we work like `getKey(key, defaultValue)` if defaultValue is not specified, then we want to use ones specified in default configs of firebaese remote, if specified use that. That is basically one use case that will enable this feature to make life easier and set defaults where needed, in the code which is fetching the config key also

Currently is impossible to know if `false` was specified in remote config, or is the default value or type default, and if we can replace it with wrapper `defaultValue=true` for example ...

Or, extend your functions `getString|getBool|getInt|...` with a second optional parameter named defaultValue, if that is specified then takes priority over default specified in `setDefaults` method.

Then would be easy like this
```
bool shouldShowModal = remoteConfig.getBool('shouldShowModal', defaultValue: true);
// if no value exist in real remote config, then true should be used and possible setDefaults for this particular key be ignored as this one has higher priority

method signature would be, so defaultValue is optional
bool getBool(String key, {bool? defaultValue = null})
```

**Comment by Jaydyzzle813** [comment]
Do what you need to do. That’s not me

**Reply by kristijorgji** [reply]
I don't understand what you mean
